components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImpl.java

```diff
@@ -902,14 +916,17 @@ private void validateOrganizationStatusUpdate(String value, String organizationI
         }
     }
 
-    private List<ExpressionNode> getExpressionNodes(String filter, String after, String before)
+    private List<ExpressionNode> getExpressionNodes(String filter, String after, String before,
+                                                    boolean isFilteringMetaAttributes)
             throws OrganizationManagementClientException {
 
         List<ExpressionNode> expressionNodes = new ArrayList<>();
         if (StringUtils.isBlank(filter)) {
             filter = StringUtils.EMPTY;
         }
-        String paginatedFilter = getPaginatedFilter(filter, after, before);
+        String paginatedFilter = isFilteringMetaAttributes ?
+                                getPaginatedFilterForMetaAttributes(filter, after, before) :
+                                getPaginatedFilter(filter, after, before);
```

Comment:
These two variations has been occurred due to the sort order of the result right?
orgs are sorted in descending order by default, but attributes are sorted in ascending order.
Hence make that property as the decision point and make the logic generalize